src/extensions/default/PhpTooling/CodeHintsProvider.js

```diff
@@ -0,0 +1,205 @@
+    var phpSuperGlobalVariables = JSON.parse(require("text!phpGlobals.json"));
+
+    function CodeHintsProvider(client) {
+        this.client = client;
+        this.query = "";
+    }
+
+    function formatTypeDataForToken($hintObj, token) {
+        $hintObj.addClass('brackets-hints-with-type-details');
+        if (token.detail) {
+            if (token.detail.trim() !== '?') {
+                if (token.detail.length < 30) {
+                    $('<span>' + token.detail.split('->').join(':').toString().trim() + '</span>').appendTo($hintObj).addClass("brackets-hints-type-details");
+                }
+                $('<span>' + token.detail.split('->').join(':').toString().trim() + '</span>').appendTo($hintObj).addClass("hint-description");
+            }
+        } else {
+            if (token.keyword) {
+                $('<span>keyword</span>').appendTo($hintObj).addClass("brackets-hints-keyword");
+            }
+        }
+        if (token.documentation) {
+            $hintObj.attr('title', token.documentation);
+            $('<span></span>').text(token.documentation.trim()).appendTo($hintObj).addClass("hint-doc");
+        }
+    }
+
+    function filterWithQueryAndMatcher(hints, query) {
+        var matchResults = $.map(hints, function (hint) {
+            var searchResult = matcher.match(hint.label, query);
+            if (searchResult) {
+                for (var key in hint) {
+                    searchResult[key] = hint[key];
+                }
+            }
+
+            return searchResult;
+        });
+
+        return matchResults;
+    }
+
+    CodeHintsProvider.prototype.hasHints = function (editor, implicitChar) {
+        if (!this.client) {
+            return false;
+        }
+
+        return true;
+    };
+
+    CodeHintsProvider.prototype.getHints = function (implicitChar) {
+        if (!this.client) {
+            return null;
+        }
+
+        var editor = EditorManager.getActiveEditor(),
+            pos = editor.getCursorPos(),
+            docPath = editor.document.file._path,
+            $deferredHints = $.Deferred(),
+            self = this;
+
+        this.client.requestHints({
+            filePath: docPath,
+            cursorPos: pos
+        }).done(function (msgObj) {
+            var context = TokenUtils.getInitialContext(editor._codeMirror, pos),
+                hints = [];
+
+            self.query = context.token.string.slice(0, context.pos.ch - context.token.start);
+            if (msgObj) {
+                var res = msgObj.items || [];
+                // There is a bug in Php Language Server, Php Language Server does not provide superGlobals
+                // Variables as completion. so these variables are being explicity put in response objects
+                // below code should be removed if php server fix this bug.
+                for(var key in phpSuperGlobalVariables) {
```

Comment:
I don't think we should add the static hints here always, otherwise we'll show hints even when they are not required. It should only come when implicitChar is '$', or otherwise filtered by the current token's prefix in case the token starts with '$'.

Reply:
it will shown only if prefix is '$'. in below code "filterWithQueryAndMatcher" will filter it.